python/lsst/dax/imgserv/getimage/imageget.py

```diff
@@ -0,0 +1,625 @@
+        unit : `str`
+            [ `pixel', 'pixels', 'arcsec' ]
+
+        Returns
+        -------
+        cutout: `afwImage.Exposure`
+
+        """
+        # Get the corresponding image(data) id from the butler
+        data_id = self.data_id_from_ccd_exp_id(ccd_exp_id)
+        if data_id:
+            # make id compatible with qResult type via custom wrapping
+            cutout = self._cutout_by_data_id(data_id, ra, dec, width, height,
+                                             unit)
+            return cutout
+
+    def cutout_from_skymap_id(self, skymap_id, filt, ra, dec, width, height,
+                              unit):
+        """
+        Parameters
+        ----------
+        skymap_id : `str`
+        filt : filter
+        ra : `float`
+        dec : `float`
+        width : `float`
+        height : `float`
+        unit : `str`
+            [ 'px', 'pix', 'pixel', 'pixels', 'arcsec']
+
+        Returns
+        -------
+        cutout: `afwImage.Exposure`
+
+        """
+        skymap = SkymapImage(self._butler, skymap_id, self._log)
+        center_coord = afwGeom.SpherePoint(ra, dec, afwGeom.degrees)
+        cutout = skymap.get(center_coord, width, height, filt, unit)
+        return cutout
+
+    def data_id_from_ccd_exp_id(self, ccd_exp_id):
+        """ The ids match the ids in _butler_keys and valid is false
+        if at least one of the ids is missing.
+
+        Parameters
+        ----------
+        ccd_exp_id: `int`
+
+        Returns
+        -------
+        data_id : `dict`
+            the list of ids derived from scienceId.
+
+        """
+        data_id = {}
+        ccd_exp_id = int(ccd_exp_id)
+        if self._butler_keys == sorted(["run", "camcol", "field", "filter"]):
+            possible_fields = {
+                "field": ccd_exp_id % 10000,
+                "camcol": (ccd_exp_id // 10000) % 10,
+                "filter": "ugriz"[(ccd_exp_id // 100000) % 10],
+                "run": ccd_exp_id // 1000000,
+            }
+            self._log.debug("data_id_from_ccd_exp_id {}".format(
+                possible_fields))
+            for key in self._butler_keys:
+                value = possible_fields[key]
+                data_id[key] = value
+            self._log.debug("dataID={}".format(data_id))
+        elif self._butler_keys == sorted(["tract", "patch", "filter"]):
+            patch_y = (ccd_exp_id // 8) % (2 ** 13)
+            patch_x = (ccd_exp_id // (2 ** 16)) % (2 ** 13)
+            possible_fields = {
+                "filter": "ugriz"[ccd_exp_id % 8],
+                "tract": ccd_exp_id // (2 ** 29),
+                "patch": "%d,%d" % (patch_x, patch_y)
+            }
+            self._log.debug("data_id_from_ccd_exp_id {}".format(
+                possible_fields))
+            for key in self._butler_keys:
+                value = possible_fields[key]
+                data_id[key] = value
+            self._log.debug("dataID={}".format(data_id))
```

Comment:
Same here, it seems like you're just copying possible_fields with the for loop on :329